src/inmanta/db/versions/v202403280.py

```diff
@@ -0,0 +1,54 @@
+"""
+    Copyright 2024 Inmanta
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+    Contact: code@inmanta.com
+"""
+
+from asyncpg import Connection
+
+
+async def update(connection: Connection) -> None:
+    """
+    Add all resource identifying columns to the resource persistent state table to improve efficiency of resource filtering and
+    sorting. These tables are derived from the resource id and are therefore part of the identity of a resource. They will not
+    change for the lifetime of a resource (with a given resource id).
+    """
+
+    await connection.execute(
+        """
+        ALTER TABLE public.resource_persistent_state
+            ADD COLUMN resource_type varchar,
+            ADD COLUMN agent varchar,
+            ADD COLUMN resource_id_value varchar
+            ;
+        UPDATE public.resource_persistent_state AS rps
```

Comment:
I wonder if we keep 'resource_persistent_state' always consistent with `resource` or if we need to trim it before doing this query?

Reply:
I think this is a good catch. If there are superorphans here, as you called them, the `NOT NULL` will be rejected. I'll rerequest review once I've fixed this.